src/forum-tool-selector-settings/ForumToolSelectorSettings.jsx

```diff
@@ -0,0 +1,150 @@
+        onMouseUp={toggleSelected}
+      >
+        {props.locked && (
+          <i className="fas fas-lock" aria-hidden="true"></i>
+        )}
+        {!props.locked && (
+          <input type="checkbox" name="checkbox" readOnly={true} className="forum-tool-checkbox form-check-input is-invalid-nodanger" checked={selected} />
+        )}
+        <div className="card-title h5"><img className="forum-tool-logo" src={props.logo} /></div>
+        <p className="card-text">{props.text}</p>
+        <p className="card-text tool-support">{props.support}</p>
+      </div>
+    </div>
+  );
+}
+
+function renderCards(forums) {
+  var components = [];
+  for (var key in forums) {
+    components.push(
+      <ForumCard
+        key={key}
+        id={key}
+        logo={forums[key].logo}
+        text={forums[key].text}
+        support={forums[key].support}
+        locked={forums[key].locked}
+      />
+    );
+  }
+  return (
+    <div className="row">
+      {components}
+    </div>
+  );
+}
+
+function renderTable(forums, features) {
+  var headings = [];
+  headings.push(
+    <th key='heading-blank'></th>
+  )
+  for (const key in forums) {
+    headings.push(
+      <th key={key}>{key}</th>
+    );
+  }
+
+  const data = [];
+  for (const key in features) {
+    const components = [];
+    components.push(
+      <td key={key}>{features[key]}</td>
+    );
+    for (const keyname in forums) {
+      if (forums[keyname].features.indexOf(key) > -1) {
+        components.push(
+          <td key={keyname} className="feature-td-center">✓</td>
+        );
+      } else {
+        components.push(
+          <td key={keyname} />
+        );
+      }
+    }
+    data.push(
+      <tr key={key}>{components}</tr>
+    );
+  }
+
+  return (
+    <table className="forum-features-table">
+      <thead>
+        <tr>{headings}</tr>
+      </thead>
+      <tbody>
+        {data}
+      </tbody>
+    </table>
+  );
+}
+
+function renderContent(props) {
+  const forumsChunked = StudioApiService.getForumToolData(props.courseId);
+  const features = StudioApiService.getForumToolFeaturesList();
+  const components = [];
+
+  for (let i = 0; i < forumsChunked.length; i++) {
+    const renderedCards = renderCards(forumsChunked[i]);
+    const renderedTable = renderTable(forumsChunked[i], features);
+    components.push(
+      <div key={i} className="selection-page">
+        <div className="card-deck">
+          {renderedCards}
+        </div>
+        <h2 className="supported-features">Supported Features</h2>
+        {renderedTable}
+      </div>
+    );
+  }
+  return components;
+}
+
+export default function ForumToolSelectorSettings(props) {
+  return (
+    <div className="container-fluid">
+      <h6 className="heading-which-tool">
+        <FormattedMessage
+          id="authoring.forumSelectorTool.whichtool"
+          defaultMessage="Which discussion tool would you like to use for this course?"
+        />
+      </h6>
+      {renderContent(props)}
```

Comment:
The way to compose JSX is to break it down into subcomponents, not functions like these.